slasher/src/database.rs

```diff
@@ -481,7 +482,11 @@ impl<E: EthSpec> SlasherDB<E> {
             .ok_or(Error::MissingIndexedAttestation {
                 id: indexed_attestation_id.as_u64(),
             })?;
-        ssz_decode(bytes)
+        // TODO(electra): make slasher fork-aware and return correct variant of attestation. Both
+        // Base and Electra variant can the same SSZ encoding, however the smaller list maximum of
+        // Base can error with an Electra attestation with heavy participation.
+        let indexed_attestation: IndexedAttestationBase<E> = ssz_decode(bytes)?;
+        Ok(IndexedAttestation::Base(indexed_attestation))
```

Comment:
Previous code on `ssz_decode` depended on the decode fallback implemented here https://github.com/sigp/lighthouse/blob/3ac3ddb2b7d4e4a824593db0a57fc6f9e39ec710/consensus/types/src/indexed_attestation.rs#L194

However, the logic is not correct. You can have an indexed attestation after the Electra fork epoch with 1 participant which will decode correctly as base. The slasher should not care about base / electra since it just solving a sets problem. It should be the duty of the consuming beacon node to produce the correct indexed type for block inclusion.

Reply:
started this here but haven't implemented the slasher db migration: https://github.com/dapplion/lighthouse/pull/35

Reply:
Merged